db/model/Variable.ts

```diff
@@ -501,9 +502,15 @@ export const fetchS3ValuesByPath = async (
     dataPath: string
 ): Promise<OwidVariableMixedData> => {
     // avoid cache as Cloudflare worker caches up to 1 hour
-    return (
-        await fetch(`${dataPath}?nocache`, { agent: httpsAgent })
-    ).json() as Promise<OwidVariableMixedData>
+    const resp = await retryPromise(() =>
+        fetch(`${dataPath}?nocache`, { agent: httpsAgent })
+    )
+    if (!resp.ok) {
```

Comment:
It's always a bit tricky because there are two ways that a request can fail: It can be a 403 or 404, which is covered by `resp.ok`.
Or it can be a networking error etc., in which case it would throw anyhow.

I'm not entirely sure here, maybe your change in `FunctionalRouter` already takes care of the second case?

Reply:
okay great!